src/build.d

```diff
@@ -121,6 +117,38 @@ Command-line parameters
     }
     foreach (target; targets.parallel(1))
         target();
+
+    writeln("Success");
+}
+
+string targetsHelp()
+{
+    string result = "";
+    foreach (dep; DependencyRange(rootDeps.map!(a => a()).array))
+    {
+        if (dep.name)
+        {
+            enum defaultPrefix = "\n                      ";
+            result ~= dep.name;
+            string prefix = defaultPrefix[1 + dep.name.length .. $];
+            void add(string msg)
+            {
+                result ~= format("%s%s", prefix, msg);
+                prefix = defaultPrefix;
+            }
+            if (dep.description)
+                add(dep.description);
+            else if (dep.targets)
```

Comment:
Can't it print the description and `target.RelativePath`?  Maybe I need to see a sample output to understand what's going on.

Reply:
You could, it's just a little more output than you might want.  For example, all of the optabgen source files would be printed.

Reply:
Here's the current output:
```
Targets
-------
dmd                   Build dmd
unittest              Run the dmd unittests
dmdconf               generated/linux/release/64/dmd.conf
lexer                 generated/linux/release/64/lexer.a
backend               generated/linux/release/64/backend.a
optabgen              Generate source files for the backend
dbackend              generated/linux/release/64/dbackend.o

```

Reply:
Here's the output if I remove the `optabgen` description:
```
Targets
-------
dmd                   Build dmd
unittest              Run the dmd unittests
dmdconf               generated/linux/release/64/dmd.conf
lexer                 generated/linux/release/64/lexer.a
backend               generated/linux/release/64/backend.a
optabgen              generated/linux/release/64/debtab.d
                      generated/linux/release/64/optab.d
                      generated/linux/release/64/cdxxx.d
                      generated/linux/release/64/elxxx.d
                      generated/linux/release/64/fltables.d
                      generated/linux/release/64/tytab.d
dbackend              generated/linux/release/64/dbackend.o

```

Reply:
Here's the output if I always print the description and targets:
```
Targets
-------
dmd                   Build dmd
unittest              Run the dmd unittests
dmdconf               generated/linux/release/64/dmd.conf
lexer                 generated/linux/release/64/lexer.a
backend               generated/linux/release/64/backend.a
optabgen              Generate source files for the backend
                      generated/linux/release/64/debtab.d
                      generated/linux/release/64/optab.d
                      generated/linux/release/64/cdxxx.d
                      generated/linux/release/64/elxxx.d
                      generated/linux/release/64/fltables.d
                      generated/linux/release/64/tytab.d
dbackend              generated/linux/release/64/dbackend.o

```

Reply:
Ok, then I guess I'd prefer that you only print the description in the `targetsHelp` function, and add a description for all dependencies.  You could, maybe, append the target to the description (e.g. `description = "generate " ~ target;`).

Reply:
>  you could, maybe, append the target to the description (e.g. description = "generate " ~ target;).

... for those that only generate a single file.

Reply:
I like being able to see what files each target is generating, just like in a Makefile.  Note that you can also provide the filename to build it:
```
./build.d dmd.conf
./build.d lexer.a
./build.d dbackend.o
```

So seeing the filename in the help is also useful if you just want to run a build command with a filename instead of a target name.

I would just print all the target files in the help, but it doesn't work for dependencies with no target files (i.e. dmd) and the optabgen can clutter up the help, so I just added the description field in cases where we don't just want to print the target files.

Reply:
I've updated the DDOC on the description field:
```
    string description; /// optional string to describe this dependency rather than printing the target files
```